torch/csrc/autograd/function.h

```diff
@@ -256,6 +256,20 @@ struct TORCH_API Function : std::enable_shared_from_this<Function> {
     return post_hooks_;
   }
 
+  // delete all post hooks matching HookType
+  template <typename HookType>
+  void delete_post_hooks() {
+    std::lock_guard<std::mutex> lock(this->mutex_);
```

Comment:
None of the other accessors use a mutex. I suppose is fine if you have a single control thread that decides when to mutate structures like this one. I think either none of them or all of them should use a mutex.